src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,312 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import argparse
+import collections
+import random
+import math
+import subprocess
+
+# Local packages
+import settings
+path = os.path.join(settings.TEMPLATE_DIR)
+sys.path.append(path)
+import validate # Function
+from constants import _constants # Constant values
+from parse_args import parse_args # Argument parser
+from wite_file import _write_file
+
+_GENERATED_FILENAME = '''binary-ops-{NUMBER}.js''' # The file name prefix of the generated tests
+_BINARY_OPERATORS = ["add", "sub", "div", "mul", "mod", "xor", "and", "or", "rightshift", "leftshift",
+"unsignedrightshift"] # The binary operators used in this script
+
+# Needed for unsigned right shift
+def unsigned32(signed):
+    return signed % 0x100000000
+
+class GenSequence:
+    def __init__(self, options, calc):
+        num_operands = options.operand_count
+        self.num_operands = num_operands
+        self.operators_list = []
+        for op in options.test_cases:
+            # Arithmetic operators
+            if (calc == 1):
+                if (op == "add"):
+                    self.operators_list.append(self.add)
+                elif (op == "sub"):
+                    self.operators_list.append(self.sub)
+                elif (op == "mul"):
+                    self.operators_list.append(self.mul)
+                elif (op == "div"):
+                    self.operators_list.append(self.div)
+                elif (op == "mod"):
+                    self.operators_list.append(self.mod)
+                else:
```

Comment:
You could use the following structure (or a similar one), and simplify not just this, but other code sequences below:
```
class A:
  def __init__(self):
    self.oper_func = {
      "add": self.add,
      "sub": self.sub,
      "mod": self.mod
    }

  def add(self):
    print("Function: add")
  def sub(self):
    print("Function: sub")
  def mod(self):
    print("Function: mod")

  def test(self, text):
    if text in self.oper_func:
      self.oper_func[text]()
    else:
      print("Not found")

a = A()

for t in ["add", "mod", "nothing", True]:
  a.test(t)
```